opal/mca/common/ucx/common_ucx.c

```diff
@@ -48,87 +48,59 @@ static void opal_common_ucx_mem_release_cb(void *buf, size_t length, void *cbdat
 
 OPAL_DECLSPEC void opal_common_ucx_mca_var_register(const mca_base_component_t *component)
 {
-    static const char *default_tls = "rc_verbs,ud_verbs,rc_mlx5,dc_mlx5,ud_mlx5,cuda_ipc,rocm_ipc";
-    static const char *default_devices = "mlx*";
-    static int hook_index;
-    static int verbose_index;
-    static int progress_index;
-    static int tls_index;
-    static int devices_index;
-    int param;
+    char *default_tls = "rc_verbs,ud_verbs,rc_mlx5,dc_mlx5,ud_mlx5,cuda_ipc,rocm_ipc";
+    char *default_devices = "mlx*";
```

Comment:
may be set default values into ```opal_common_ucx``` definition? at line 38 above
then evaluation ```if (NULL == opal_common_ucx.tls)``` below may be omitted

Reply:
like
```
opal_common_ucx_module_t opal_common_ucx =
 {
     .progress_iterations = 100,
     .opal_mem_hooks = 1,
     .tls="rc_verbs,ud_verbs,rc_mlx5,dc_mlx5,ud_mlx5,cuda_ipc,rocm_ipc",
     .devices="mlx*"
 };```

Reply:
Deregistration will set it back to NULL. With strings you have to set the value before each registration call. In this case the variable may be actively registered so don't want to set it if it is already set.